src/argilla/client/feedback/dataset/base.py

```diff
@@ -349,19 +359,26 @@ def prepare_for_training(
                 " dataset via the `FeedbackDataset.add_records` method first."
             )
 
-        if isinstance(task_mapping, TrainingTaskMappingForTextClassification):
-            self.unify_responses(question=task_mapping.label.question, strategy=task_mapping.label.strategy)
-        else:
-            raise ValueError(f"Training data {type(task_mapping)} is not supported yet")
-
-        data = task_mapping._format_data([record for record in self.records])
+        if isinstance(task, TrainingTaskForTextClassification):
+            self.unify_responses(question=task.label.question, strategy=task.label.strategy)
+        elif not isinstance(
+            task,
+            (
+                TrainingTaskForSupervisedFinetuning,
+                TrainingTaskForRewardModelling,
+                TrainingTaskForDirectPreferenceOptimization,
+            ),
+        ):
+            raise ValueError(f"Training data {type(task)} is not supported yet")
+
+        data = task._format_data(self)
```

Comment:
I would like to simplify the code below this line, when checking for the frameworks and the methods required to prepare the data

Reply:
👍 Do you want to do that as a follow-up PR?

Reply:
Sure, we can have a look at it upon merge! 👍🏻